x-pack/plugins/triggers_actions_ui/public/application/hooks/use_rule_aad_fields.ts

```diff
@@ -8,21 +8,67 @@
 import { DataViewField } from '@kbn/data-views-plugin/common';
 import { useKibana } from '@kbn/kibana-react-plugin/public';
 import { BASE_RAC_ALERTS_API_PATH } from '@kbn/rule-registry-plugin/common';
-import useAsync from 'react-use/lib/useAsync';
-import type { AsyncState } from 'react-use/lib/useAsync';
+import { HttpSetup } from '@kbn/core/public';
+import { useQuery } from '@tanstack/react-query';
+import { i18n } from '@kbn/i18n';
+import { useMemo } from 'react';
 import { TriggersAndActionsUiServices } from '../..';
 
-export function useRuleAADFields(ruleTypeId?: string): AsyncState<DataViewField[]> {
-  const { http } = useKibana<TriggersAndActionsUiServices>().services;
+const EMPTY_AAD_FIELDS: DataViewField[] = [];
 
-  const aadFields = useAsync(async () => {
-    if (!ruleTypeId) return [];
-    const fields = await http.get<DataViewField[]>(`${BASE_RAC_ALERTS_API_PATH}/aad_fields`, {
-      query: { ruleTypeId },
-    });
+async function fetchAadFields({
+  http,
+  ruleTypeId,
+}: {
+  http: HttpSetup;
+  ruleTypeId?: string;
+}): Promise<DataViewField[]> {
+  if (!ruleTypeId) return EMPTY_AAD_FIELDS;
+  const fields = await http.get<DataViewField[]>(`${BASE_RAC_ALERTS_API_PATH}/aad_fields`, {
+    query: { ruleTypeId },
+  });
+
+  return fields;
+}
+
+export function useRuleAADFields(ruleTypeId?: string): {
+  aadFields: DataViewField[];
+  loading: boolean;
+} {
+  const {
+    http,
+    notifications: { toasts },
+  } = useKibana<TriggersAndActionsUiServices>().services;
+
+  const queryAadFieldsFn = () => {
+    return fetchAadFields({ http, ruleTypeId });
+  };
+
+  const onErrorFn = () => {
+    toasts.addDanger(
+      i18n.translate('xpack.triggersActionsUI.useRuleAADFields.errorMessage', {
+        defaultMessage: 'Unable to load alert fields per rule type',
+      })
+    );
+  };
 
-    return fields;
+  const {
+    data: aadFields = EMPTY_AAD_FIELDS,
+    isInitialLoading,
+    isLoading,
+  } = useQuery({
+    queryKey: ['loadAlertAadFieldsPerRuleType'],
+    queryFn: queryAadFieldsFn,
+    onError: onErrorFn,
+    refetchOnWindowFocus: false,
+    enabled: ruleTypeId !== undefined,
   });
 
-  return aadFields;
+  return useMemo(
+    () => ({
+      aadFields,
+      loading: ruleTypeId === undefined ? false : isInitialLoading || isLoading,
```

Comment:
I think we can just use `isLoading` since `enabled` should prevent this from turning `true` if `ruleTypeId` is undefined

Reply:
I thought that too, but I got loading true even with enabled false